aries-site/src/layouts/navigation/Search.js

```diff
@@ -1,114 +1,180 @@
-import React, { useContext, useRef, useEffect, useState } from 'react';
+import { useContext, useState } from 'react';
 import { useRouter } from 'next/router';
-import styled from 'styled-components';
 import PropTypes from 'prop-types';
-import { Box, Button, TextInput, ResponsiveContext, Keyboard } from 'grommet';
-import { Search as SearchIcon } from 'grommet-icons';
+import { Layer, ResponsiveContext } from 'grommet';
 import { getSearchSuggestions, nameToPath } from '../../utils';
 import { internalLink } from '../../components';
+import { SearchResult, SearchResults } from '.';
 
 const allSuggestions = getSearchSuggestions;
 
-// Using Search icon as the arialabelledby for the text input. Documentation
-// on why this is a valid replacement for using label here:
-// https://www.w3.org/WAI/tutorials/forms/labels/#using-aria-labelledby
-// https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/Basic_form_hints
-const StyledTextInput = styled(TextInput).attrs(() => ({
-  'aria-labelledby': 'search-icon',
-}))``;
+/*
+ * Construct an object where 'label' is the displayed element and
+ * 'value' contains all data related to the content/result.
+ */
+const formatSearchResults = (query, results) =>
+  results &&
+  results.map(result => {
+    const nextValue = {
+      ...result.value,
+      url: nameToPath(result.value.title),
+    };
 
-export const Search = ({ focused, setFocused }) => {
+    return {
+      label: <SearchResult query={query} result={nextValue} />,
+      value: nextValue,
+    };
+  });
+
+/* Sort search results by relevancy and/or strength of match type */
+const sortSearch = matches => {
+  const results = matches
+    /* Prioritize title matches over metadata matches */
```

Comment:
this would also prioritize title matches over content matches?

Reply:
TBD. Not necessarily. I think this sorting is temporary and can be enhanced quite a bit. I have a concept of a "relevancy score" which should drive the ultimate sort order, but that's a future scope of work.

Reply:
sounds good this was more curiosity on my end